### ISSUE TITLE
Brew upgrade: from php 7.2.x with xdebug to 7.2.4 now throws "Host is down" errors

### ISSUE THREAD

**Issue by dpsarrou** [Homebrew/homebrew-core#26111]
- [x ] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [ x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? 
- [x ] ran `brew update` and can still reproduce the problem?
- [x ] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why):
Os: High Sierra 10.13.4
I was running php 7.2.x with xdebug (installed by homebrew since February 25 if I remember correctly) without any problems. 
Yesterday I ran `brew upgrade`. Since then my locally installed php app running on 7.2.4 throws errors, specifically when using PDO but I've also managed to reproduce this using curl from php.

- What happened (include command output)
Running my typical test suite from codeception, after the update I receive the following error "Db: SQLSTATE[HY000] [2002] Host is down while creating PDO connection ". Of course I have checked and the host is not down."

- What you expected to happen:
No error should be thrown and everything should work properly just like before.

- Step-by-step reproduction instructions (by running `brew install` commands)
I was running php 7.2.x (I don't remember what version exactly; from the file system I see this was done on February 25. Therefore someone could cross-reference the version that brew would ship from that).
I run `brew upgrade`. This installed php 7.2.4:
```
$ php -v
PHP 7.2.4 (cli) (built: Mar 29 2018 15:19:46) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.4, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v2.6.0, Copyright (c) 2002-2018, by Derick Rethans
```
As a PoC I tried to make an http call from that php. The PoC did succeed in reproducing the error:

```
php -r 'echo file_get_contents("https://google.com");'

PHP Warning:  file_get_contents(https://google.com): failed to open stream: Host is down in Command line code on line 1
PHP Stack trace:
PHP   1. {main}() Command line code:0
PHP   2. file_get_contents() Command line code:1

Warning: file_get_contents(https://google.com): failed to open stream: Host is down in Command line code on line 1

Call Stack:
    0.0004     383584   1. {main}() Command line code:0
    0.0005     383584   2. file_get_contents() Command line code:1

```

I ran brew doctor, brew upgrade, brew update and brew reinstall php. The error persists. 
Then I tried `brew uninstall php`. This reverts to:
```
$ php -v
PHP 7.1.14 (cli) (built: Feb  7 2018 18:33:30) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2018 Zend Technologies
```
Running again the above PoC `php -r 'echo file_get_contents("https://google.com");'`
This time I receive the html page, as expected.

I can see that this version of php does not have xdebug installed. The error "Host is down" is very cryptic and I do not directly see the relation with xdebug - pdo - curl. I do not have sufficient knowledge of the internals however so I wouldnt be surprised.

Fact is that I cannot run my php app anymore (the code requires php 7.2 features) using the brew php. Does someone have the solution or at least a workaround to suggest in the mean time..? 
Next thing I will try is to install xdebug manually, although I have been postponing this because the whole benefit of using brew is to make installation/removal of packages as easy as an one liner `brew install php`.

**Comment by SMillerDev** [comment]
Installing of packages should be done with pecl anyway after the migration to the core tap. As for the connection issue, what does `brew doctor` say?

**Comment by dpsarrou** [comment]
Actually I just tried installing xdebug via pecl. I was surprised to see the same error in there as well:
```
$ pecl search xdebug
Connection to `pecl.php.net:80' failed: Host is down
```

Below is the output of brew doctor:
```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libtcl8.6.dylib
  /usr/local/lib/libtk8.6.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/fakemysql.h
  /usr/local/include/fakepq.h
  /usr/local/include/fakesql.h
  /usr/local/include/itcl.h
  /usr/local/include/itcl2TclOO.h
  /usr/local/include/itclDecls.h
  /usr/local/include/itclInt.h
  /usr/local/include/itclIntDecls.h
  /usr/local/include/itclMigrate2TclCore.h
  /usr/local/include/itclTclIntStubsFcn.h
  /usr/local/include/mysqlStubs.h
  /usr/local/include/node/ares.h
  /usr/local/include/node/ares_version.h
  /usr/local/include/node/nameser.h
  /usr/local/include/node/node_internals.h
  /usr/local/include/node/pthread-fixes.h
  /usr/local/include/node/uv-private/ngx-queue.h
  /usr/local/include/node/uv-private/stdint-msvc2008.h
  /usr/local/include/node/uv-private/tree.h
  /usr/local/include/node/uv-private/uv-bsd.h
  /usr/local/include/node/uv-private/uv-darwin.h
  /usr/local/include/node/uv-private/uv-linux.h
  /usr/local/include/node/uv-private/uv-sunos.h
  /usr/local/include/node/uv-private/uv-unix.h
  /usr/local/include/node/uv-private/uv-win.h
  /usr/local/include/node/v8-preparser.h
  /usr/local/include/node/v8stdint.h
  /usr/local/include/odbcStubs.h
  /usr/local/include/pqStubs.h
  /usr/local/include/tcl.h
  /usr/local/include/tclDecls.h
  /usr/local/include/tclOO.h
  /usr/local/include/tclOODecls.h
  /usr/local/include/tclPlatDecls.h
  /usr/local/include/tclThread.h
  /usr/local/include/tclTomMath.h
  /usr/local/include/tclTomMathDecls.h
  /usr/local/include/tdbc.h
  /usr/local/include/tdbcDecls.h
  /usr/local/include/tdbcInt.h
  /usr/local/include/tk.h
  /usr/local/include/tkDecls.h
  /usr/local/include/tkPlatDecls.h

Warning: Unbrewed .pc files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected .pc files:
  /usr/local/lib/pkgconfig/tcl.pc
  /usr/local/lib/pkgconfig/tk.pc

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libtclstub8.6.a
  /usr/local/lib/libtkstub8.6.a

Warning: You have unlinked kegs in your Cellar
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  python@2
  gnupg

```

**Comment by SMillerDev** [comment]
All I could find for that error is this by the way, which suggests it's not a PHP issue: https://aminsblog.wordpress.com/2012/05/12/mac-os-x-host-is-down-with-curl-python-pipetc/

**Comment by dpsarrou** [comment]
I went through various articles as well but I was paying less attention to old posts as I thought they wouldn't be related to php 7.2.4. Sir, you indeed are correct, Little Snitch was the culprit. It automatically blocks the new php binary complaining about its code signature. Maybe this is something for you to think about as well..? In any case with this I managed to resolve the issues. Thanks for the quick help!
